### ISSUE TITLE
Ranges for color space values

### ISSUE THREAD

**Issue by MichaelMauderer** [gtaylor/python-colormath#43]
I just noticed that color spaces do not all document the ranges of expected values and are somewhat inconsistent, e.g., `XYZColor` seems to assume values scaled from 0...1 and `LChab` assumes values from 0...100 (for L and C). The current implementation of the CAMs then again expects XYZ values  from 0...100.

So at the very least this should be documented everywhere, and then maybe things should be made consistent in the future (but this is of course a backwards compatibility problem).

I will definitely adjust the CAMs, so they don't produce unexpected behavior when used with the values from `XYZColor` objects.

**Comment by gtaylor** [comment]
I just released 2.1.0, which had a good deal of new stuff in it. With that landing out in the wild, we should start getting some feedback on those changes.

That'll free us up to work on this range re-jiggering, which is going to be pretty disruptive. Perhaps enough to jump to 3.x.

**Comment by pixelou** [comment]
The range for a and b in Lab color space is not documented either. Could anyone indicated the ranges here until it gets fixed in the documentation please?

**Comment by fnorf** [comment]
I came here looking for the same. What are the L, a, b ranges? I see L values roughly between 1000 and 6000, a and b from -2000 to 4000 so I am completely lost.

**Comment by chalmersgit** [comment]
I came here with the same question about Lab ranges. I can't find anything on it in the documentation.

According to Wikipidia, L = [0,100], a = [-128,127] and b = [-128,127]
https://en.wikipedia.org/wiki/CIELAB_color_space

But it also says that this is depends on the specific implementation, and those above values are usually for 8-bit precision (colormath supports 16-bit).

Using the example:
https://python-colormath.readthedocs.io/en/latest/delta_e.html#example

I tried messing around with various numbers (such as above 100 or below 0 for L, but no warning was shown).

Can anyone let me know what these ranges are?

**Comment by earthbound19** [comment]
I also wonder.

**Comment by escalonn** [comment]
This issue affected me too. The values given by `colormath` agree empirically with the ones at [this converter site online](https://www.nixsensor.com/free-color-converter/), suggesting that this project uses L = [0,100], a = [-128,128] and b = [-128,128].

**Comment by EmielStoelinga** [comment]
Any updates about the range of LAB values that is expected by the library?

**Comment by KelSolaar** [comment]
CIE Lab does not impose limits besides that of the Standard Observer. HDR colours, even though CIE Lab is not well suited to represent them, can go way beyond **L=100**, and the **a** and **b** are only bounded by the Visible Spectrum and will change depending  on your current working RGB colourspace as seen in this notebook cell: https://colab.research.google.com/drive/1NRcdXSCshivkwoU2nieCvC3y14fx1X4X#scrollTo=_Ggl1OAuST-B

**Comment by kannes** [comment]
I don't think that is the point of this issue, at least it wasn't to me. What we need is documentation on how the range is scaled. I.e. is it 0..1 or 0..100 for the common range of L?

PS: Your notebook requires a log-in.

**Comment by KelSolaar** [comment]
There are 10 participants to this issue asking different questions. **L*** should be 0-100 for practical purposes, **a** and **b**, don't really have practical ones, 0-100 is fine, even though they will go higher.

The notebook should be readable without sign-in, I just tried on 2 different computers without being logged in them. Same is true when using the incognito mode.
